common/logfield.hpp

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_LOGFIELD_HPP
+#define MENDER_LOGFIELD_HPP
+
+namespace mender::log {
+
+using namespace std;
+
+struct LogField {
+	LogField(string key, const string value) :
+		key(move(key)),
```

Comment:
I think `move` is ill-formed on a non-rvalue-reference. You'd need a dedicated move constructor, but then you also need a regular constructor for when you can't use an rvalue reference.

I think in general we should not worry about move semantics so much unless we have actually measured poor performance, and stick to either const-references, copy-by-value, or passing shared pointers, depending on the situation. Safety first!

Reply:
> Unit tests? slightly_smiling_face

Coming, as soon as we're happy with the interface.

Reply:
https://github.com/mendersoftware/mender/pull/1120#discussion_r1066664375